bookingmanagerpersistence/src/main/java/cz/fi/muni/pa165/entity/User.java

```diff
@@ -96,4 +104,48 @@ public void setRole(Role role) {
     public void setReservations(Set<Reservation> reservations) {
         this.reservations = reservations;
     }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Equals by mel byt Business equivalence, jak jsme ho delali na cviceni, takze fieldy bez ID a pristupovat k nim pres get metody.

Reply:
Já to nechával nagenereovat IDE.

Reply:
Já to generoval IDE. Takhle může být:     @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (obj == null)
            return false;
        if (!(obj instanceof User))
            return false;
        User other = (User) obj;
        if (fullName == null) {
            if (other.getFullName() != null)
                return false;
        } else if (!fullName.equals(other.getFullName()))
            return false;
        if (email == null) {
            if (other.getEmail() != null)
                return false;
        } else if (!email.equals(other.getEmail()))
            return false;
        if (address == null) {
            if (other.getAddress() != null)
                return false;
        } else if (!address.equals(other.getAddress()))
            return false;
        if (email == null) {
            if (other.getEmail() != null)
                return false;
        } else if (!email.equals(other.getEmail()))
            return false;
        if (phoneNumber == null) {
            if (other.getPhoneNumber() != null)
                return false;
        } else if (!phoneNumber.equals(other.getPhoneNumber()))
            return false;
        if (passwordHash == null) {
            if (other.getPasswordHash() != null)
                return false;
        } else if (!passwordHash.equals(other.getPasswordHash()))
            return false;
        if (role == null) {
            if (other.getRole() != null)
                return false;
        } else if (!role.equals(other.getRole()))
            return false;
        
        return true;
    }

Reply:
I přístup k vlastním fieldum by měl být přes get metodu. Takže místo např fullName==null by mělo být getFullName()==null